### PR TITLE
Re-add integration testing to CI workflow

### PR DESCRIPTION
**Pull request by NaomiReeves**
Somehow this was dropped so I'm re-adding it to the workflow and updating some of the other action steps in our CI workflow.

The upstream source for the action step to run the [rubocop linter as it's own step has been archived](https://github.com/andrewmcodes/rubocop-linter-action), so I'm cleaning that up.

### CODE REVIEW

**Review by jaymzh — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,24 +23,14 @@ jobs:
       run: bundle exec rspec
     - name: Run rubocop
       run: bundle exec rubocop --display-cop-names
-# This doesn't work on PRs from forks due to GITHUB_TOKEN, which makes
```
Comment (jaymzh) [inline .github/workflows/ci.yml:26]: I'd leave this... I think they've fixed this now and I don't wanna re-remember how to do this :)

Comment (NaomiReeves) [reply .github/workflows/ci.yml:26]: This wasn't fixed on the upstream repo and as I note in my summary, [it was ultimately archived as a result of it]( https://github.com/andrewmcodes/rubocop-linter-action). Or do you want to just see the attempt to do this?

Comment (jaymzh) [reply .github/workflows/ci.yml:26]: it wasn't an issue with upstream it was a limitation in how Actions which, which has been fixed.

File: test/run-broken.sh [hunk 2]
```diff
@@ -15,7 +15,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+set -e
 set -o verbose
+bundle exec chef-zero &
+ZERO=$!
+trap "kill -9 $ZERO" EXIT
```
Comment (davide125) [inline test/run-broken.sh:22]: Do we really want to `-9` here? This probably warrants a comment either way (same for the two other scripts).

Comment (jaymzh) [reply test/run-broken.sh:22]: -9 seems right. we're exiting/being killed, we 100% want chef-zero to due immediately. Plus it's all in-memory so there's no danger.

Comment (NaomiReeves) [reply test/run-broken.sh:22]: I'm really just mirroring what we have in [grocery-delivery](https://github.com/facebook/grocery-delivery/blob/master/test/run-git.sh) for these tests.

File: test/run-broken.sh [hunk 3]
```diff
@@ -15,14 +15,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+set -e
 set -o verbose
-rm -rf /tmp/ops
+bundle exec chef-zero &
+ZERO=$!
+trap "kill -9 $ZERO" EXIT
+SOURCE=/tmp/ops
```
Comment (davide125) [inline test/run-broken.sh:23]: This should probably use `mktemp -d` for safety

**Review by davide125 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jaymzh) [reply .github/workflows/ci.yml:26]: ugh, that's sad becasue it actually works now, but yeah kill it.

File: test/run-broken.sh (hunk 3 above)
Comment (NaomiReeves) [reply test/run-broken.sh:23]: So the scripts that this calls (ie: `tt-auto.rb` and `tt-git.rb`) are referencing the specific repo `/tmp/ops` and while I can probably feed the temp dir to this, I'm not sure that I want to tackle that here too.

**Pull request closed without merging**